qiskit_nature/problems/second_quantization/lattice/models/ising_model.py

```diff
@@ -0,0 +1,114 @@
+            uniform_onsite_potential: The on-site potential. (or uniform_external_field)
+            onsite_interaction: The strength of the on-site interaction, the Ising model does not
+                have an on-site interaction.
+
+        Returns:
+            The Ising model with uniform parameters.
+
+        Raises:
+            Warning: If the on-site interaction is not None.
+        """
+        if onsite_interaction is not None:
+            raise Warning(
+                "The Ising model does not have on-site interactions. Provided onsite-interaction "
+                "parameter will be ignored."
+            )
+        return super().uniform_parameters(lattice, uniform_hopping, uniform_onsite_potential, None)
+
+    @classmethod
+    def from_parameters(
+        cls, hopping_matrix: np.ndarray, onsite_interaction: complex = None
+    ) -> "IsingModel":
+        """Return the Hamiltonian of the Ising model from the given hopping matrix.
+
+        Args:
+            hopping_matrix: A real or complex valued square symmetric matrix.
+            onsite_interaction: The strength of the on-site interaction, the Ising model does not
+                have an on-site interaction.
+
+        Returns:
+            IsingModel: The Ising model generated from the given hopping matrix.
+
+        Raises:
+            Warning: If the on-site interaction is not None.
+            ValueError: If the hopping matrix is not square matrix, it is invalid.
+        """
+        if onsite_interaction is not None:
+            raise Warning(
+                "The Ising model does not have on-site interactions. Provided onsite-interaction "
+                "parameter will be ignored."
+            )
+        return super().from_parameters(hopping_matrix, None)
+
+    def second_q_ops(self, display_format: Optional[str] = None) -> SpinOp:
+        """Return the Hamiltonian of the Ising model in terms of `SpinOp`.
+
+        Args:
+            display_format: Not supported for Spin operators.
+
+        Returns:
+            SpinOp: The Hamiltonian of the Ising model.
+
+        Raises:
+            Warning: If display-format is not None.
+        """
+        if display_format is not None:
+            raise Warning(
+                "Spin operators do not support display-format. Provided display-format "
+                "parameter will be ignored."
+            )
+        ham = []
+        weighted_edge_list = self._lattice.weighted_edge_list
+        register_length = self._lattice.num_nodes
+        # kinetic terms
+        for node_a, node_b, weight in weighted_edge_list:
+            if node_a == node_b:
+                index = node_a
+                ham.append((f"X_{index}", weight))
+
+            else:
+                index_left = node_a
+                index_right = node_b
+                hopping_parameter = weight
+                ham.append((f"Z_{index_left} Z_{index_right}", hopping_parameter))
```

Comment:
In the case of Ising model the `hopping_parameter` name looks a bit strange. Would it be possible to rename it to `coupling_parameter`. This holds also for the rest of the model. 

```suggestion
                ham.append((f"Z_{index_left} Z_{index_right}", coupling_parameter))
```